sha1-name.c

```diff
@@ -1773,12 +1773,12 @@ static int get_oid_with_context_1(const char *name,
 			if (new_filename)
 				filename = new_filename;
 			if (flags & GET_OID_FOLLOW_SYMLINKS) {
-				ret = get_tree_entry_follow_symlinks(&tree_oid,
+				ret = get_tree_entry_follow_symlinks_gently(&tree_oid,
 					filename, oid, &oc->symlink_path,
-					&oc->mode);
+					&oc->mode, flags & GET_OID_GENTLY);
 			} else {
-				ret = get_tree_entry(&tree_oid, filename, oid,
-						     &oc->mode);
+				ret = get_tree_entry_gently(&tree_oid, filename, oid,
+						     &oc->mode, flags & GET_OID_GENTLY);
```

Comment:
It would make sense to invert the commit order, I guess, to make lower-level functions optionally gentle first, and only then use that in subsequent commits to make callers more gentle.

(I know, this is not the way it was developed, but it sounds more logical to me, as a reader of the patches.)